src/PyCall.jl

```diff
@@ -36,6 +31,25 @@ import Base.Iterators: filter
 include(joinpath(dirname(@__FILE__), "..", "deps","depsutils.jl"))
 include("startup.jl")
 
+# Python C API is not interrupt-safe.  In principle, we should
+# use sigatomic for every ccall to the Python library, but this
+# should really be fixed in Julia (#2622).  However, we will
+# use the `disable_sigint` function to protect *all* invocations
+# of Python API.  This is required since `pyjlwrap_call` uses
+# `reenable_sigint` which has to be called within `disable_sigint`
+# context.
+macro ccall(args...)
+    quote
+        disable_sigint() do
```

Comment:
This adds a higher-order function call to every `ccall`; have you checked whether it makes a measurable difference in performance?  (Hopefully it all gets inlined, but…)

I really wish this would get fixed in Julia itself.

Reply:
Wait, https://github.com/JuliaLang/julia/issues/2622 is closed, does that mean we don't need this anymore?

Reply:
Oh, now I vaguely remember https://github.com/JuliaLang/julia/pull/16174 — we still need this for any `ccall` in that might call back to Julia code in which we want to catch interrupt exceptions, I think?

Reply:
The comment should clarify this, in any case.

Reply:
You mean a comment like this?:

```
Although #2622 is closed, ccall still does not automatically disable SIGINT.
```

> have you checked whether it makes a measurable difference in performance?

No, but the code without it was unsafe.  It manifested easily because I added `reenable_sigint` so that `julia` runtime complains.  Wasn't it needed to avoid setfault like #15?

Having said that, there are places where we can merge `disable_sigint` and do it in a batch while using `ccall`s.  If we detect performance regression I guess we can use such tricks.  I need to look at `benchmarks/` code to do the check.